### PR TITLE
CLM-6663 Add documentation for graphQL write for discussion board

### PR DESCRIPTION
**Pull request by AlexONeillTI**

### CODE REVIEW

File: definition-map.js [hunk 1]
```diff
@@ -193,6 +193,73 @@ const MUTATIONS = {
             completedAssessmentAttemptId: "The ID of the last completed assessment attempt.",
         }
     },
+    CreateThread: {
+        metadescription: "Creates a thread of comments under a commentable type.",
+        args: {
+            forumId: "The ID of the forum.",
```
Comment (rjschill87) [inline definition-map.js:199]: I wouldn't think you'd have to manually add these as they should be covered by this block, right?

https://github.com/thoughtindustries/helium-graphql/blob/57b346dae70b4244efb6611efc8eabe724ecedc2/augment-schema.js#L83-L85

File: definition-map.js [hunk 2]
```diff
@@ -193,6 +193,73 @@ const MUTATIONS = {
...
+            courseId: "The ID of the course.",
+            clientId: "The ID of the client.",
+            layoutId: "The ID of the layout.",
+            widgetTitle: "The title of the widget.",
+            commentableType: "The type of thread requested, based on the forum's location, e.g. discussion board, assignment, widget thread etc",
```
Comment (rjschill87) [inline definition-map.js:204]: Wouldn't be based on the type requested, as the user would be creating a thread, so "The type of thread to create..."

File: definition-map.js [hunk 3]
```diff
@@ -193,6 +193,73 @@ const MUTATIONS = {
...
+            asset: "The URL of the asset.",
+            assetFileName: "The file name of the asset.",
+            videoAsset: "The ID of the video asset.",
+            body: "The body of the comment.",
+            title: "The title of the comment.",
+            notificationsEnabled: "Flag to enable notifications."
+        }
+    },
+    UpdateThread: {
+        metadescription: "Adds an additional comment into an exisiting thread",
```
Comment (rjschill87) [inline definition-map.js:214]: This mutation would update the opening comment of the thread, not create an additional one.

File: definition-map.js [hunk 4]
```diff
@@ -193,6 +193,73 @@ const MUTATIONS = {
...
+        args: {
+            id: "The ID of the thread",
```
Comment (rjschill87) [inline definition-map.js:216]: Missing periods/consistency with `args` descriptions above.

File: definition-map.js [hunk 5]
```diff
@@ -193,6 +193,73 @@ const MUTATIONS = {
...
+            body: "The body of the comment",
+            title: "The title of the comment",
+            asset: "The URL of the asset.",
+            commentableType: "The type of thread requested, based on the forum's location, e.g. discussion board, assignment, widget thread etc"
```
Comment (rjschill87) [inline definition-map.js:220]:
```suggestion
            commentableType: "The type of thread to be updated, based on the forum's location, e.g. discussion board, assignment, widget thread etc"
```

File: definition-map.js [hunk 6]
```diff
@@ -193,6 +193,73 @@ const MUTATIONS = {
...
+        }
+    },
+    DestoyThread: {
+        metadescription: "Removed a thread from a commentableType",
```
Comment (rjschill87) [inline definition-map.js:224]:
```suggestion
        metadescription: "Deletes a thread and associated comments.",
```

File: definition-map.js [hunk 7]
```diff
@@ -193,6 +193,73 @@ const MUTATIONS = {
...
+        args: {
+            id: "The ID of the thread",
+            commentableType: "The type of thread requested, based on the forum's location, e.g. discussion board, assignment, widget thread etc"
```
Comment (rjschill87) [inline definition-map.js:227]:
```suggestion
            commentableType: "The type of thread to be deleted, based on the forum's location, e.g. discussion board, assignment, widget thread etc"
```

File: definition-map.js [hunk 8]
```diff
@@ -193,6 +193,73 @@ const MUTATIONS = {
...
+        args: {
+            id: "The ID of the id",
+            body: "The updated body of the comment",
+            asset: "The updated asset of the comment",
+            assetFileName: "The file name of the asset.",
+            commentableType: "The type of thread requested, based on the forum's location, e.g. discussion board, assignment, widget thread etc"
+        }
+    },
+    DestroyComment: {
+        metadescription: "Removed a comment from a commentableType",
```
Comment (rjschill87) [inline definition-map.js:241]:
```suggestion
        metadescription: "Removed a comment from a thread",
```

File: definition-map.js [hunk 9]
```diff
@@ -193,6 +193,73 @@ const MUTATIONS = {
...
+        }
+    },
+    UpdateComment: {
+        metadescription: "Updates the body and/or asset of an existing comment",
+        args: {
+            id: "The ID of the id",
+            body: "The updated body of the comment",
+            asset: "The updated asset of the comment",
+            assetFileName: "The file name of the asset.",
+            commentableType: "The type of thread requested, based on the forum's location, e.g. discussion board, assignment, widget thread etc"
```
Comment (rjschill87) [inline definition-map.js:237]: "The type of the comment's parent thread to be updated"

File: definition-map.js [hunk 10]
```diff
@@ -193,6 +193,73 @@ const MUTATIONS = {
...
+        }
+    },
+    DestroyComment: {
+        metadescription: "Removed a comment from a commentableType",
+        args: {
+            id: "The ID of the comment",
+            commentableType: "The type of thread requested, based on the forum's location, e.g. discussion board, assignment, widget thread etc"
```
Comment (rjschill87) [inline definition-map.js:244]: Same re: this being the type of the comment's parent thread

File: definition-map.js [hunk 11]
```diff
@@ -193,6 +193,73 @@ const MUTATIONS = {
...
+        }
+    },
+    CreateCommentLike: {
+        metadescription: "Creates a comment on a commentableType",
```
Comment (rjschill87) [inline definition-map.js:248]: Doesn't create a comment on a commentableType so much as adds a like to a comment.

File: definition-map.js [hunk 12]
```diff
@@ -193,6 +193,73 @@ const MUTATIONS = {
...
+        args: {
+            commentId: "The ID of the comment", 
+            commentableId: "The ID of the commentable entity.",            
+            commentableType: "The type of thread requested, based on the forum's location, e.g. discussion board, assignment, widget thread etc",
+             clientId: "The ID of the client"
+        }
+    },
+    RemoveCommentLike: {
+        metadescription: "Removed a comment from a commentableType",
```
Comment (rjschill87) [inline definition-map.js:257]: Same, only removing instead of adding.

**Comment by AlexONeillTI** [comment]
All review comments addressed. Also modified some of the descriptions regarding the "comments location" or "threads location" instead of "forums location". Would this be correct?

File: definition-map.js [hunk 13]
```diff
@@ -177,7 +177,7 @@ const MUTATIONS = {
         metadescription: "Creates a comment.",
         args: {
             commentableId: "The ID of the commentable entity.",
-            commentableType: "The type of the commentable entity.",
+            commentableType: "The type of comment to create, based on the comment's location, e.g. discussion board, assignment, widget thread etc.",
```
Comment (rjschill87) [inline definition-map.js:180]: This wouldn't really be the comment's type - it's the type of the comment's parent thread.

File: definition-map.js [hunk 14]
```diff
@@ -192,6 +192,65 @@ const MUTATIONS = {
             completedAssessmentAttemptId: "The ID of the last completed assessment attempt.",
         }
     },
+    CreateThread: {
+        metadescription: "Creates a thread of comments under a commentable type.",
+        args: {
+            widgetTitle: "The title of the widget.",
+            commentableType: "The type of thread to create, based on the comment's location, e.g. discussion board, assignment, widget thread etc.",
```
Comment (rjschill87) [inline definition-map.js:199]: based on the thread's*

File: definition-map.js [hunk 15]
```diff
@@ -192,6 +192,65 @@ const MUTATIONS = {
...
+            body: "The body of the comment.",
+            title: "The title of the comment.",
+            notificationsEnabled: "Flag to enable notifications."
+        }
+    },
+    UpdateThread: {
+        metadescription: "Updates the opening comment of the thread.",
+        args: {
+            id: "The ID of the thread.",
+            body: "The body of the comment.",
```
Comment (rjschill87) [inline definition-map.js:212]: I would clarify that these are the "opening comment of the thread"

File: definition-map.js [hunk 16]
```diff
@@ -192,6 +192,65 @@ const MUTATIONS = {
...
+        args: {
+            id: "The ID of the comment.",
+            commentableType: "The type of the comment's parent thread to destory, based on the forum's location, e.g. discussion board, assignment, widget thread etc."
+        }
+    },
+    CreateCommentLike: {
+        metadescription: "Adds a like to a comment.",
+        args: {
+            commentableId: "The ID of the commentable entity.",            
+            commentableType: "The type of comment, based on the comment's location, e.g. discussion board, assignment, widget thread etc.",
```
Comment (rjschill87) [inline definition-map.js:246]: Would be 'thread' again here as well.

File: definition-map.js [hunk 17]
```diff
@@ -192,6 +192,65 @@ const MUTATIONS = {
...
+        }
+    },
+    RemoveCommentLike: {
+        metadescription: "Removed a like from a comment.",
+        args: {
```
Comment (rjschill87) [inline definition-map.js:251]: Since `args` is empty we can just remove it.

**Review by rjschill87 — APPROVED** [review]

**Pull request merged**